pkg/network/tracer/tracer_test.go

```diff
@@ -504,6 +504,37 @@ func TestTCPCollectionDisabled(t *testing.T) {
 	require.False(t, ok)
 }
 
+func TestTCPConnsReported(t *testing.T) {
+	// Setup
+	config := testConfig()
+	config.CollectTCPConns = true
+
+	tr, err := NewTracer(config)
+	require.NoError(t, err)
+	defer tr.Stop()
+
+	server := NewTCPServer(func(c net.Conn) {})
+	doneChan := make(chan struct{})
+	err = server.Run(doneChan)
+	require.NoError(t, err)
+	defer close(doneChan)
+
+	// Connect to server
+	c, err := net.DialTimeout("tcp", server.address, 50*time.Millisecond)
+	require.NoError(t, err)
+	defer c.Close()
+
```

Comment:
you should probably wait the request been processed by the server  <-doneChan

Reply:
I am not sure what you mean? `doneChan` is only for stopping the server, it is not aware of ongoing requests as far as I know.

Reply:
yeah, it's the other way ...
I mean when the connection is done, close it in the message handler and create another chan to wait in the test that the message/connection has been processed

Reply:
Should be fixed in d1460d758977dae38b74d8b6607a44af7aca921c